OIIInspector/image_manager.py

```diff
@@ -0,0 +1,198 @@
+import logging
+from retry import retry
+import subprocess
+import time
+import shlex
+from OIIInspector.exceptions import AddressAlreadyInUse, OIIInspectorError
+from OIIInspector.utils import run_cmd
+
+log = logging.getLogger(__name__)
+
+
+class ImageManager:
+    """
+    Context manager for working wight images
+    """
+    _grpc_start_port = 50051
+    _grpc_max_port_tries = 100
+    _grpc_init_wait_time = 5
+    _base_container_name = "OIIInspector_running_image"
+    _image_is_running = False
+    _container_platform = None
+
+    def __init__(self, image_address):
+        """
+        Initialization of the ImageManager
+
+        :param image_address:
+        :type image_address:
+        """
+        self._get_container_platform()
+        self._image_address = None
+        self._port = self._grpc_start_port
+        self._rpc_proc = None
+        self._image_address = image_address
+
+    def __enter__(self):
+        self.start_image()
+        return self
+
+    def __exit__(self, exc_type, exc_value, exc_tb):
+        self.close_image_manager()
+
+    def start_image(self):
+        """
+        Method that starts desired image
+        """
+        # pull needs to be done separately, because command run (which is capable of pull), would be terminated by timer
+        self._pull_image()
+        self._port, self._rpc_proc = self._serve_index_registry()
+        self._image_is_running = True
+
+    def get_local_address_of_image(self) -> str:
```

Comment:
Could you change this into a property? 
https://www.programiz.com/python-programming/property
Also rename it to "grpcurl_address" or something like that